fabric-tool-attribute-api-v1/src/main/java/net/fabricmc/fabric/impl/tool/attribute/handlers/ModdedToolsVanillaBlocksToolHandler.java

```diff
@@ -55,18 +56,18 @@ private ToolItem getVanillaItem(int miningLevel) {
 	@NotNull
 	@Override
 	public ActionResult isEffectiveOn(Tag<Item> tag, BlockState state, ItemStack stack, LivingEntity user) {
-		if (stack.getItem() instanceof DynamicAttributeTool) {
+		if (stack.getItem() instanceof DynamicAttributeTool tool) {
 			if (ToolManagerImpl.entryNullable(state.getBlock()) != null) {
 				// Block is a modded block, and we should ignore it
 				return ActionResult.PASS;
 			}
 
 			// Gets the mining level from our modded tool
-			int miningLevel = ((DynamicAttributeTool) stack.getItem()).getMiningLevel(tag, state, stack, user);
+			int miningLevel = tool.getMiningLevel(tag, state, stack, user);
 			if (miningLevel < 0) return ActionResult.PASS;
 
-			ToolItem vanillaItem = getVanillaItem(miningLevel);
-			return vanillaItem.isSuitableFor(state) ? ActionResult.SUCCESS : ActionResult.PASS;
+			FabricTool vanillaItem = (FabricTool) getVanillaItem(miningLevel);
+			return vanillaItem.isSuitableFor(miningLevel, state) ? ActionResult.SUCCESS : ActionResult.PASS;
```

Comment:
This doesn't really need to be changed to use the interface's `isSuitableFor` since they return the same value because of the mixin, but this is fine too I guess.

Reply:
`getVanillaItem` can only return... well, vanilla items. That is, for example, wooden, stone, iron, diamond, and netherite pickaxes, of mining levels 0, 1, 2, 3, and 4 respectively.
If `getVanillaItem(5)` is called, it will return the highest one it knows about, which is netherite, which has a mining level of _4_ and not the desired 5.
Thus, if the original `isSuitableFor(state)` were to be called on the item the method returns, in this case it would be checking if a tool with a mining level of _4_ was suitable for the state... which is not what we want.
So, instead, we use `isSuitableFor(miningLevel, state)` to ensure that we are indeed checking for a mining level of 5.